src/components/SignUp.js

```diff
@@ -0,0 +1,112 @@
+import React from 'react'
+import { Grid, Paper, Avatar, TextField, Button, Typography } from '@material-ui/core'
+import AddCircleOutlineOutlinedIcon from '@material-ui/icons/AddCircleOutlineOutlined';
+import { Formik, Form, Field, ErrorMessage} from 'formik';
+import * as Yup from 'yup';
+import { useNavigate} from 'react-router-dom';
+
+
+export function SignUp() {
+
+    const navigate = useNavigate();
+
+    const fields = [
+        {
+            id: 1,
+            label: "First name",
+            name: "firstName",
+            placeholder: "Enter first name",
+            type: "text"
+        },
+        {
+            id: 2,
+            label: "Last name",
+            name: "lastName",
+            placeholder: "Enter last name",
+            type: "text"
+        },
+        {
+            id: 3,
+            label: "Email",
+            name: "email",
+            placeholder: "Enter email name",
+            type: "email"
+        },
+        {
+            id: 4,
+            label: "Password",
+            name: "password",
+            placeholder: "Enter password",
+            type: "password"
+        },
+        {
+            id: 5,
+            label: "Confirm password",
+            name: "confirmPassword",
+            placeholder: "Confirm password",
+            type: "password"
+        }
+    ];
+   
+    const initialValues = {
+        firstName: '',
+        lastName: '',
+        email: '',
+        password: '',
+        confirmPassword: ''
+    }
+
+    const validationSchema = Yup.object().shape({
+        firstName: Yup.string().required("Required").matches(/^[A-Za-z]+$/, 'This field cannot contain white space and special characters'),
+        lastName: Yup.string().required("Required").matches(/^[A-Za-z]+$/, 'This field cannot contain white space and special characters'),
+        email: Yup.string().email('Please enter valid email').required("Required"),
+        password: Yup.string().min(6, "Password must contain at least 6 characters").required("Required"),
```

Comment:
Password must contain at least one upper case, one special character and one number